<launch>
  <arg name = "mode_str" default = "localization" />
  <arg name = "control_type" default = "CarFollowing" />
  <arg name = "course_file" default = "$(find controller_calc)/Path_Info/Test_resampled_path2.csv" />
  <arg name = "direction" default = "true" />
  <arg name = "number_of_car" default = "2" />
  <arg name = "id_0" default = "1010" /> 
  <arg name = "id_1" default = "1011" />
  <arg name = "id_2" default = "12" />
  <arg name = "id_3" default = "13" />
  <arg name = "manual_0" default = "true" />
  <arg name = "manual_1" default = "false" />
  <arg name = "manual_2" default = "false" />
  <arg name = "manual_3" default = "false" />
  <arg name = "idx_0" default = "0" /> 
  <arg name = "idx_1" default = "1" />
  <arg name = "using_sensor_flag" default = "false" />
  <arg name = "distance_radar_gain" default = "1.0" />
  <arg name = "using_custom_topic_flag" default = "false" />
  <arg name = "state_packet_leader" default = "CarFollowing_Car_0/state_packet" />
  <arg name = "state_packet_follower" default = "CarFollowing_Car_1/state_packet" />

  <group ns = "Car_Following">
    <node pkg = "cacc_controller" type = "state_estimator" name = "state_estimator" output="screen">
      <param name = "position_type_str" value = "$(arg mode_str)" />
      <param name = "control_type" value = "$(arg control_type)" />
      <param name = "course_file" value = "$(arg course_file)" />
      <param name = "direction" value = "$(arg direction)" />
      <param name = "number_of_car" value = "$(arg number_of_car)" />
      <param name = "id_0" value = "$(arg id_0)" />
      <param name = "id_1" value = "$(arg id_1)" />
      <param name = "id_2" value = "$(arg id_2)" />
      <param name = "id_3" value = "$(arg id_3)" />
      <param name = "manual_0" value = "$(arg manual_0)" />
      <param name = "manual_1" value = "$(arg manual_1)" />
      <param name = "manual_2" value = "$(arg manual_2)" />
      <param name = "manual_3" value = "$(arg manual_3)" />
      <param name = "distance_radar_gain" value = "$(arg distance_radar_gain)" />
    </node>

    <node pkg = "cacc_controller" type = "following_controller" name = "following_controller" output="screen">
      <param name = "idx_0" value = "$(arg idx_0)" />
      <param name = "idx_1" value = "$(arg idx_1)" />
      <param name = "id_0" value = "$(arg id_0)" />
      <param name = "id_1" value = "$(arg id_1)" />
      <param name = "id_2" value = "$(arg id_2)" />
      <param name = "id_3" value = "$(arg id_3)" />
      <param name = "using_sensor_flag" value = "$(arg using_sensor_flag)" />
      <param name = "using_custom_topic_flag" value = "$(arg using_custom_topic_flag)" />
      <param name = "state_packet_leader" value = "$(arg state_packet_leader)" />
      <param name = "state_packet_follower" value = "$(arg state_packet_follower)" />
    </node>
  </group>

</launch>

